pkg/resolver/searchSubscription.go

```diff
@@ -0,0 +1,55 @@
+// Copyright Contributors to the Open Cluster Management project
+package resolver
+
+import (
+	"context"
+	"time"
+
+	klog "k8s.io/klog/v2"
+
+	"github.com/stolostron/search-v2-api/graph/model"
+	"github.com/stolostron/search-v2-api/pkg/config"
+)
+
+func SearchSubscription(ctx context.Context, input []*model.SearchInput) (<-chan []*SearchResult, error) {
+	ch := make(chan []*SearchResult)
+
+	// You can (and probably should) handle your channels in a central place outside of `schema.resolvers.go`.
+	// For this example we'll simply use a Goroutine with a simple loop.
+	go func() {
+		timeout := time.After(time.Duration(config.Cfg.SubscriptionPollTimeout) * time.Minute)
+		// Handle deregistration of the channel here. Note the `defer`
+		defer close(ch)
+
+		for {
+			klog.V(3).Info("Search subscription new poll interval")
+			searchResult, err := Search(ctx, input)
+
+			if err != nil {
+				klog.Errorf("Error occurred during the search subscription request: %s", err)
+			}
+
+			// The subscription may have been closed due to the client disconnecting.
+			// Hence we do send in a select block with a check for context cancellation.
+			// This avoids goroutine getting blocked forever or panicking,
+			select {
+			case <-ctx.Done(): // This runs when context gets cancelled. Subscription closes.
+				klog.V(3).Info("Search subscription Closed")
+				return // Remember to return to end the routine.
+
+			case <-timeout: // This runs when timeoout is hit. Subscription closes.
+				klog.V(3).Info("Subscription timeout reached. Closing connection.")
+				return // Remember to return to end the routine.
+			
+			case ch <- searchResult: // This is the actual send.
+				// Our message went through, do nothing 
```

Comment:
I want to take a close look at what is actually being sent. We don't want to send the entire searchResult object every 10 seconds because it will be a lot of unnecessary data. We just want to send the delta.
I hope the library implementation takes care of it, but I'd like to confirm it.

Reply:
With the current logic we would be sending the entire search result back. I will need to look into getting just the delta, and how that would need to be handled in frontend as well.

Reply:
Let's talk about this one. This is good progress, but I'm thinking we must resolve this soon to avoid having to re-do the front-end changes.

Reply:
Ok, from the function I see we can only add more []*SearchResult to the channel, so this would be the entire object.  This will also cause issues if there's multiple SearchInput in the subscription. All of the inputs will need to get evaluated.

```
func SearchSubscription(ctx context.Context, input []*model.SearchInput) (<-chan []*SearchResult, error) {
	ch := make(chan []*SearchResult)
```

Let's continue learning. :-)